faiss/MetricType.h

```diff
@@ -31,11 +31,19 @@ enum MetricType {
     METRIC_Canberra = 20,
     METRIC_BrayCurtis,
     METRIC_JensenShannon,
+    METRIC_Jaccard, ///< defined as: sum_i(min(a_i, b_i)) / sum_i(max(a_i, b_i))
+                    ///< where a_i, b_i > 0
 };
 
 /// all vector indices are this type
 using idx_t = int64_t;
 
+/// this function is used to distinguish between min and max indexes since
+/// we need to support similarity and dis-similarity metrics in a flexible way
+constexpr bool is_dissimilairty_metric(MetricType metric_type) {
+    return metric_type == METRIC_INNER_PRODUCT;
```

Comment:
Hmm it should be `is_similarity_metric` in fact, since larger values of the metric are better...
Also, Jaccard is also a similarity metric no?